src/components/ImageModal/index.tsx

```diff
@@ -0,0 +1,351 @@
+  const [currentImageId, setCurrentImageId] = useState<string>(firstImageId)
+  const clientX = useRef<number | null>(null)
+  const topElement = useMemo(() => document.createElement('div'), [])
+
+  useEffect(() => {
+    document.body.append(topElement)
+
+    return () => {
+      document.body.removeChild(topElement)
+    }
+  }, [])
+
+  useEffect(() => {
+    getImagesInfo()
+  }, [images])
+
+  useEffect(() => {
+    setCurrentImageId(firstImageId)
+  }, [isOpen])
+
+  const getImagesInfo = async (): Promise<void> => {
+    const fulfilledImages = images.map(({ src, id }) => {
+      const image = new Image()
+      image.src = src
+
+      return new Promise(resolve => {
+        image.onload = (): void => {
+          resolve({
+            height: image.height,
+            id,
+            src,
+            width: image.width * (DEFAULT_RESIZE_HEIGHT / image.height)
+          })
+        }
+      })
+    })
+
+    imagesInfo.current = (await Promise.all(
+      fulfilledImages
+    )) as ResizeImageInfo[]
+  }
+
+  const getSumImageWidth = (): number =>
+    [...imagesInfo.current].reduce(
+      (sumImageWidth, currentImage, idx, images) => {
+        const { id, width } = currentImage
+
+        if (currentImageId === id) {
+          // for early return
+          images.splice(1)
+          return sumImageWidth + width / 2 + IMAGE_GAP_HALF
+        }
+
+        return sumImageWidth + width + IMAGE_GAP_HALF
+      },
+      0
+    )
+
+  const getImageCount = (): number => {
+    return imagesInfo.current.findIndex(({ id }) => currentImageId === id)
+  }
+
+  const handleClickIndicator = (id: string): void => {
+    setCurrentImageId(id)
+  }
+
+  const handleTouchStart: TouchEventHandler = (e): void => {
+    if (clientX.current === null) {
+      clientX.current = e.changedTouches[0].pageX
+    }
+  }
+
+  const handleTouchEnd: TouchEventHandler = (e): void => {
+    const previousTouchX = clientX.current
+    const currentTouchX = e.changedTouches[0].pageX
+    const currentImageIdx = getImageCount()
+
+    if (previousTouchX === null) {
+      return
+    }
+
+    const goNextImage =
+      previousTouchX - currentTouchX > TOUCH_START_END_DIFFERENCE
+    const goPreviousImage =
+      currentTouchX - previousTouchX > TOUCH_START_END_DIFFERENCE
+
+    if (goNextImage) {
+      const nextImageIndex = currentImageIdx + 1
+      const nextImage = imagesInfo.current[nextImageIndex]
+      const newCurrentImageId = nextImage
+        ? nextImage.id
+        : imagesInfo.current[0].id
+
+      setCurrentImageId(newCurrentImageId)
+      return
+    }
+
+    if (goPreviousImage) {
+      const prevImageIdx = currentImageIdx - 1
+      const prevImage = imagesInfo.current[prevImageIdx]
+      const newCurrentImageId = prevImage
+        ? prevImage.id
+        : imagesInfo.current[imagesInfo.current.length - 1].id
+
+      setCurrentImageId(newCurrentImageId)
+      return
+    }
+
+    clientX.current = null
+  }
```

Comment:
```suggestion
const NAV_TYPE = {
    NEXT: 'next',
    PREV: 'prev'
  } as const

  const VALUE_OF_NAV_TYPE = {
    next: 1,
    prev: -1
  } as const

  const handleTouchEnd: TouchEventHandler = (e): void => {
    const images = imagesInfo.current
    const prevTouchX = clientX.current
    const currentTouchX = e.changedTouches[0].pageX
    const currentImageIdx = getImageCount()

    const isNotMoved = !prevTouchX || prevTouchX < TOUCH_START_END_DIFFERENCE
    if (isNotMoved) {
      clientX.current = null
      return
    }

    const isNext = prevTouchX - currentTouchX > TOUCH_START_END_DIFFERENCE
    const navType = isNext ? NAV_TYPE.NEXT : NAV_TYPE.PREV

    const newCurrentImage = images[currentImageIdx + VALUE_OF_NAV_TYPE[navType]]
    const lastImageId = isNext ? images[0].id : images[images.length - 1].id
    const newCurrentImageId = newCurrentImage?.id || lastImageId
    setCurrentImageId(newCurrentImageId)
  }
```
next, prev에 대한 코드가 미세하게 중복이 있는데 위와 같이 변경하는 건 어떠신가요?!

Reply:
[요기](https://github.com/price-offer/offer-design-system/pull/35/commits/e90f6d96972c4a90227154e46f0ae5ee2f7b2333)에 반영 완료 😘